src/pages/TriggerReleasePlanPage.tsx

```diff
@@ -1,20 +1,66 @@
 import * as React from 'react';
 import { Helmet } from 'react-helmet';
+import { useParams } from 'react-router-dom';
+import { useK8sWatchResource } from '@openshift/dynamic-plugin-sdk-utils';
+import { Bullseye, Spinner } from '@patternfly/react-core';
 import NamespacedPage from '../components/NamespacedPage/NamespacedPage';
 import PageAccessCheck from '../components/PageAccess/PageAccessCheck';
 import { TriggerReleaseFormPage } from '../components/ReleaseService/ReleasePlan/TriggerRelease/TriggerReleaseFormPage';
 import { FULL_APPLICATION_TITLE } from '../consts/labels';
-import { ReleaseModel } from '../models';
+import { ReleaseModel, ReleasePlanGroupVersionKind } from '../models';
+import ErrorEmptyState from '../shared/components/empty-state/ErrorEmptyState';
+import { HttpError } from '../shared/utils/error/http-error';
 import { AccessReviewResources } from '../types';
+import { ReleasePlanKind } from '../types/coreBuildService';
+import { useWorkspaceInfo } from '../utils/workspace-context-utils';
 
 const TriggerReleasePlanPage: React.FC<React.PropsWithChildren<unknown>> = () => {
+  const { name } = useParams();
+  const { namespace } = useWorkspaceInfo();
+
+  const [releasePlan, loaded, loadErr] = useK8sWatchResource<ReleasePlanKind>({
```

Comment:
So I can see you are sending release plan name as a parameter in URL and then fetching the complete object here which is then passed on to the next component to use application name from it. Why can't you just send the application name directly in URL param and use it?

Reply:
Do we actually need to check the releaseplan exist or not? Most of the use cases will be around triggering the release from the releaseplan action menu which means the plan actually exist. And how did the form work earlier without sending releaseplan as prop to `TriggerReleaseFormPage`